src/sentry/scripts/similarity/index.lua

```diff
@@ -220,8 +220,10 @@ end
 -- Key Generation
 
 local function get_key_prefix(configuration, index)
+    -- NB: The brackets around the scope allow redis cluster to shard keys
+    -- using the value within the brackets, this is known as a redis hash tag.
     return string.format(
-        '%s:%s:%s',
+        '%s:{%s}:%s',
```

Comment:
Out of curiosity, do these braces remain in the key forever or are they removed by something in the cluster?
 What happens if the key happens to contain them incidentally (e.g. `'\x7b\x7d'` from byte packing `[123, 125]` into `>bb`?)

Reply:
This is more of a note to self, but we should move the `scope` parameter out of `ARGV` and use `KEYS[1]` instead. I wouldn't do it here because it will increase conflicts with GH-5979.

Reply:
The hash-tag is well defined in the redis cluster spec: [key hash tags](https://redis.io/topics/cluster-spec#keys-hash-tags).

So your example of `[123,125]` being packed would be *okay* since there is nothing in-between. However, yes, any incidental `{...}` values will be used for the hash tag, *assuming* that it is the first occurrence. 

The braces stay in the key forever.

Reply:
Ah, interesting — thanks. It seems like as a general rule we'd always want to put the tag at or near the beginning of the key, then?

Reply:
Yeah, I think that's a good rule